### PR TITLE
Add documentation for new license verification

### LINKED ISSUE #53145: Update docs with this information

### PR DESCRIPTION
**Pull request by pjlast**
Closes #53145 

Doc updates

## Test plan

Just doc updates

### CODE REVIEW

File: doc/admin/licensing/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Licensing
+
+[Sourcegraph Enterprise](../../getting-started/oss-enterprise.md) requires a valid license key to enable many Enterprise-specific features.
```
Comment (erzhtor) [inline doc/admin/licensing/index.md:3]: Not related, but why the file is called `oss-enterprise`?  I though oss and enterprise are kinda different tiers.

File: doc/dev/background-information/adding_ping_data.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This page outlines the process for adding or changing the data collected from So
 
 ## Ping philosophy
 
-Pings are the only data Sourcegraph receives from installations. Our users and customers trust us with their most sensitive data. We must preserve and build this trust through only careful additions and changes to pings.
+Pings, alongside [license verification chekcs](../../admin/licensing/index.md), are the only data Sourcegraph receives from installations. Our users and customers trust us with their most sensitive data. We must preserve and build this trust through only careful additions and changes to pings.
```
Comment (erzhtor) [inline doc/dev/background-information/adding_ping_data.md:7]:
```suggestion
Pings, alongside [license verification checks](../../admin/licensing/index.md), are the only data Sourcegraph receives from installations. Our users and customers trust us with their most sensitive data. We must preserve and build this trust through only careful additions and changes to pings.
```

**Review by erzhtor — APPROVED** [review]
👍

File: doc/admin/licensing/index.md (hunk 1 above)
Comment (pjlast) [reply doc/admin/licensing/index.md:3]: Yeah the doc talks about the difference between OSS and Enterprise tiers, and what features the Enterprise tiers have. Just thought it was a good doc to link to that explains Enterprise

**Review by kopancek — APPROVED** [review]
Approving to unblock, but left some suggestions.

**Comment by eseliger** [comment]
Just saw this, I'm curious (and I'm sure readers of this are, too) - what happens when short term network blips occur, or when sourcegraph.com is not responsive for a moment? We have no SLA for that instance and if it would break all the sourcegraph instances out there within 10 minutes of the outage beginning, that sounds scary :D I'm sure you put mechanisms in place to prevent that, but it would be very reassuring to read about these in this docs page.

**Pull request merged**